### ISSUE TITLE
display depth of submerged rocks in nautical style

### ISSUE THREAD

**Issue by quantenschaum** [osmandapp/OsmAnd#18092]
# 🚀 feature request

### Description

When a `seamark:type=rock` has an attached `depth` tag, the depth value should be displayed next to the rock symbol since this information is very important.

### Describe the solution you'd like

see above

### Describe alternatives you've considered

no alternative

**Comment by miramikes** [comment]
+1
#14975
![stones](https://github.com/osmandapp/OsmAnd/assets/623412/5b0420dd-24a2-4b08-a460-7a7c922c558e)

**Comment by quantenschaum** [comment]
Exactly. Here a direct comparison of 

how it currently looks in OsmAnd  
![Screenshot_20230905-080620](https://github.com/osmandapp/OsmAnd/assets/3985599/ba3c61a9-f142-40c8-89d1-f41df0bedf31)

and how it should look like (and used to look on paper charts), a tiny cross without background is sufficient  
![Screenshot_20230905-080627](https://github.com/osmandapp/OsmAnd/assets/3985599/847f06e4-018b-44af-aec5-728d322916e6)

The rock symbols are far too big and clutter the entire chart. To be able to judge if the rock is dangerous or not to a particular vessel, the depth value must be displayed.

Note: The use of a map magnifier does not affect the size of the rock symbols, it effectively only changes the threshold (zoom level) when rocks get displayed.

**Comment by quantenschaum** [comment]
I tried to find the parts in the code that are responsible for rendering the rock. I did not find them. Could someone point me to the right spot, please?

**Comment by miramikes** [comment]
> I tried to find the parts in the code that are responsible for rendering the rock. I did not find them. Could someone point me to the right spot, please?

Not sure if this what you are asking ...
Part of marine.render.xml

      
      
      <case minzoom="13" tag="seamark:type" value="rock" icon="seamark_rock_submerged" shield="seamark_rock_shield" iconVisibleSize="15">
        <case additional="seamark:rock:water_level=covers" icon="seamark_rock_covers"/>
        <case additional="seamark:rock:water_level=awash" icon="seamark_rock_awash"/>
      </case>

Maybe also this might be useful:
https://github.com/osmandapp/OsmAnd-resources/blob/a3c976f67abe1c878efb133ef7f7bec567e8997a/icons/tools/Generating_OSMAND_map_icons.txt

**Comment by quantenschaum** [comment]
Yes, these lines I did find. But how can I change the size of the icon?

**Comment by quantenschaum** [comment]
setting `shield="null"` removes the circle background. `iconVisibleSize` only sets the effective size, such that only one overlapping icons is displayed.

**Comment by quantenschaum** [comment]
w/o the shield it already looks less cluttered

![Screenshot_20230905-223948](https://github.com/osmandapp/OsmAnd/assets/3985599/9fef5ee3-a7bc-48fe-9236-d831bfd02335)
![Screenshot_20230905-223943](https://github.com/osmandapp/OsmAnd/assets/3985599/30068e2b-1248-457e-b014-c7de39783b6b)

**Comment by miramikes** [comment]
I found this commit "Increase size of barrier=debris icon":
https://github.com/osmandapp/OsmAnd-resources/commit/1b9b7a684bff03a5eb043fb83fc9aa3dbf86c28c
Is it really necessary to change svg icon to change it's rendered size?

**Comment by quantenschaum** [comment]
Unfortunately yes AFAIK. There seems to be no setting in the render.xml that allows changing the icon size.

**Comment by quantenschaum** [comment]
The shield (circle around the rock symbol) is used in paper charts to mark rocks *dangerous to surface navigation* or *outside the corresponding depth area*.

INT1 section K
![image](https://github.com/osmandapp/OsmAnd/assets/3985599/15a04b09-3356-4abc-828e-669857bdc1df)

**Comment by miramikes** [comment]
> side question concerning depth data: Is there some documentation on how to generate an OBF containing depth contours? #12502

I used phyghtmap program to generate contours file when I builded my own maps for garmin years ago.
http://katze.tfiu.de/projects/phyghtmap/

According to man pages it can use GeoTiff file and generat pbf/xml file
pbf/xml file should be able to convert in OsmAndMapCreator to obf file
http://katze.tfiu.de/projects/phyghtmap/phyghtmap.1.html
https://osmand.narkive.com/KmMkzQIN/making-srtm-obf-files

Never tried this program to generate depht contour but I used it a lot before for contours

You will need some depth data ... maybe from here?
https://www.gebco.net/data_and_products/gridded_bathymetry_data/#global

**Comment by quantenschaum** [comment]
Thanks! Just for the record, I managed to create an OBF containing depth contours and spot soundings from data leeched from https://www.geoseaportal.de/mapapps/resources/apps/navigation (currently down :disappointed:). The workflow is roughly like this:
- download data as GeoJSON
- import into QGIS
- filter the relevant rows
- add columns
  - `depth=VALDCO`
  - `contour=depth`
  - `contourtype=[5m,10m,20m,50m,100m]` with formula using `depth%type==0`
- export layer as geopackage only containing the 3 added columns
- import into JOSM via OpenData plugin
- save as OSM xml
- convert OSM to OBF using OsmAndMapCreator

Terrible, no command line tools, but it works.

OsmAnd does not display contours without a `contourtype` tag or values of `contourtype<5m`, so a 2m line will not show up on the map. I tweaked the [nautical map style](https://github.com/quantenschaum/mapping/raw/master/marine.render.xml) and the renderer [settings for contour lines](https://github.com/quantenschaum/mapping/raw/master/depthcontourlines.addon.render.xml). Use these, and the map will look like this.

![Screenshot_20230907-131703](https://github.com/osmandapp/OsmAnd/assets/3985599/6cc69478-f604-488c-9763-eb6fe7c3cbbe)

**Comment by quantenschaum** [comment]
This can be implemented easily with https://github.com/osmandapp/OsmAnd/discussions/18116#discussioncomment-6945803

**Comment by vshcherb** [comment]
I think subscript values are better to display with textSize2 i.e. instead of "4.3" -> "4 (3)" so then it could be a separate text tag value

**Comment by quantenschaum** [comment]
There is no `textSize2` according to https://osmand.net/docs/technical/osmand-file-formats/osmand-rendering-style/.

**Comment by vshcherb** [comment]
There is nameTag2, it suggests that for 1 object 2 texts needs to be displayed, so it's automatically puts 1 text in brackets, in future we could have a feature to say ```nameTag2="subscript" textPlacement="subscript"```, now we already have textPlacement but it's for main tag (top/bottom/center) which could be later reused for 2nd text

**Comment by quantenschaum** [comment]
I still do not understand how this actually works. The [docs](https://osmand.net/docs/technical/osmand-file-formats/osmand-rendering-style/) are very short and hard to understand. 

So, what exactly needs to be in `rendering_types.xml` (T) and `render.xml` (S) for text to be displayed? How do `nameTag`, `nameTag2` in (S) and `nameTags` in (T) work?

Currently there is 

- `<type tag="seamark:type" value="rock" minzoom="10" nameTags="seamark:name"/>` in (T)
- `<type tag="depth" additional="text" minzoom="6" order="128"/>` in (T)

Simply adding `<case minzoom="10" tag="seamark:type" value="rock" textSize="14"  textDy="-12" />` to the text section in (S) makes the depth labels appear on the map, even though `depth` is never listed in any `nameTag` attribute. Why? 

What is the `order` attribute in (T) and (S) good for?

**Comment by quantenschaum** [comment]
note: In ENCs there is the attribute `EXPSOU` with contains information about the depth of the rock relative to its surrounding area. This corresponds to [`depth:exposition`](https://wiki.openstreetmap.org/wiki/Key:depth#Exposition_(EXPSOU)) in OSM. The `seamark_rock_shield` should only be displayed if `depth:exposition=shoaler`.

**Comment by quantenschaum** [comment]
With [`depth:exposition`](https://github.com/quantenschaum/mapping/blob/master/rendering_types.xml#L96) and the shield only rendered when [`shoaler`](https://github.com/quantenschaum/mapping/blob/master/marine.render.xml#L3359), it may look like this.

Rocks outside (shoaler) of the surrounding depth contour area get the shield.

![image](https://github.com/osmandapp/OsmAnd/assets/3985599/218b1dc9-032e-4870-b83f-f595b6ae620d)
